daemon/api_model.go

```diff
@@ -69,3 +89,96 @@ func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 
 }
+
+// helper for parsing url options into formatting option vars
+func parseHeadersFormatOptionsFromURL(url *url.URL) (
+	jsonResult bool,
+	headersOnly bool,
+	headers map[string]string,
+	err error,
+) {
+	headers = map[string]string{}
+	q := url.Query()
+	for k := range q {
+		if k == "json" {
+			switch q.Get(k) {
+			case "false":
+				jsonResult = false
+			case "headers":
+				headersOnly = true
+				fallthrough
+			case "true":
+				jsonResult = true
+			default:
+				err = errors.New(`"json" query parameter when used must be set to "true" or "headers"`)
+				return
+			}
+			continue
+		}
+		headers[k] = q.Get(k)
+	}
+
+	return
+}
+
+// getModel gets the current model assertion using the DeviceManager
+func getModel(c *Command, r *http.Request, _ *auth.UserState) Response {
+	jsonResult, headersOnly, _, err := parseHeadersFormatOptionsFromURL(r.URL)
+	if err != nil {
+		return BadRequest(err.Error())
+	}
+
+	devmgr := c.d.overlord.DeviceManager()
+	if devmgr == nil {
+		return InternalError("accessing device manager failed")
+	}
+
+	model, err := devmgr.Model()
+	if err != nil {
```

Comment:
we should special case state.ErrNoState, that means no model yet

Reply:
What exactly should we do in the case of no model yet? Should we loop waiting for a model?

Reply:
in `cmd/snap` perhaps, or a clear error message. In the daemon, an appropriate 404 with an ad-hoc error kind.

Reply:
yes, we want dedicated kind "device-assertion-not-available" I think, see response.go etc about how to go about introducing them

Reply:
Value would then be set to the assertion type "model"/"serial" possibly

Reply:
Okay I think I have implemented this as requested. There is now an additional errorKind in the daemon/response.go file which is used in the response we send if err is state.ErrNoState. The response we send is also a special crafted errorResult with Value set as requested, message same as what it was, and Kind as the new errorKind in daemon/response.go.